lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -419,6 +419,13 @@ LinearLayout sharedToLinearLayoutLeadingOffset(ArrayRef<int64_t> shape,
   int tileRows = 8;
   int tileCols = 8 * tileWidthBytes / elemBitWidth;
 
+  if (shape[colDim] < tileCols || shape[rowDim] < tileRows) {
+    llvm::errs() << "Illegal shared layout; expected shape to be at least ["
+                 << tileRows << ", " << tileCols << "], shape: ["
+                 << shape[rowDim] << ", " << shape[colDim] << "]\n";
+    llvm::report_fatal_error("Illegal shared layout");
+  }
```

Comment:
sorry missed this review. One question: could this be part of the verifier?

Reply:
it would be more memdesc type verifier (I believe we have some of those already).

>Maybe we can just emit an error in SharedEncodingAttr's builder
it depends on the memdesc shape which wouldn't be known in the builder right?

Reply:
I'm referring to this builder interface. https://github.com/triton-lang/triton/blob/f808819c18e118e0df415a8caaee3a0ed126f4c8/include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td#L430

Does it look good to you?

Reply:
ah yeah I guess that works. It wouldn't catch other cases but that's the only place where we create those anyway